mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/ZipEntryIterator.java

```diff
@@ -0,0 +1,80 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup.zip;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZipEntryIterator implements Iterator<ZipEntry>, Closeable {
+    private final ZipInputStream zipInputStream;
+    private Optional<ZipEntry> next;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZipEntryIterator.class);
+
+    public ZipEntryIterator(ZipInputStream inputStream) {
+        zipInputStream = inputStream;
+        try {
+            next = Optional.ofNullable(zipInputStream.getNextEntry());
+        } catch (IOException e) {
+            //EMPTY STREAM
```

Comment:
how can you know?

Reply:
It's the more common case, but effectively i can add a log